solidity/ecdsa/contracts/libraries/EcdsaAuthorization.sol

```diff
@@ -15,25 +15,474 @@
 pragma solidity ^0.8.9;
 
 import "@keep-network/sortition-pools/contracts/SortitionPool.sol";
-import "@threshold-network/solidity-contracts/contracts/staking/TokenStaking.sol";
+import "@threshold-network/solidity-contracts/contracts/staking/IStaking.sol";
 
+/// @notice Library managing the state of stake authorizations for ECDSA
+///         operator contract and the presence of operators in the sortition
+///         pool based on the stake authorized for them.
 library EcdsaAuthorization {
-    struct Data {
+    struct Parameters {
+        // The minimum authorization required by ECDSA application so that
+        // operator can join the sortition pool and do the work.
         uint96 minimumAuthorization;
+        // Authorization decrease delay in seconds between the time
+        // authorization decrease is requested and the time the authorization
+        // decrease can be approved. It is always the same value, no matter if
+        // authorization decrease amount is small, significant, or if it is
+        // a decrease to zero.
         uint64 authorizationDecreaseDelay;
     }
 
+    struct AuthorizationDecrease {
+        uint96 decreasingBy; // amount
+        uint64 decreasingAt; // timestamp
+    }
+
+    struct Data {
+        Parameters parameters;
+        mapping(address => address) stakingProviderToOperator;
+        mapping(address => address) operatorToStakingProvider;
+        mapping(address => AuthorizationDecrease) pendingDecreases;
+    }
+
+    event OperatorRegistered(
+        address indexed stakingProvider,
+        address indexed operator
+    );
+
+    event AuthorizationIncreaseRequested(
+        address indexed stakingProvider,
+        address indexed operator,
+        uint96 fromAmount,
+        uint96 toAmount
+    );
+
+    event AuthorizationDecreaseRequested(
+        address indexed stakingProvider,
+        address indexed operator,
+        uint96 fromAmount,
+        uint96 toAmount,
+        uint64 decreasingAt
+    );
+
+    event AuthorizationDecreaseApproved(address indexed stakingProvider);
```

Comment:
They do though I think coding the client could be easier if we have them on `WalletRegistry`. Ideally, the ECDSA client code wouldn't even care about `TokenStaking` existence and just reacted to `WalletRegistry` events. These particular events are important to the client because it may want to update the sortition pool state based on them.

Reply:
I'm not 100% convinced, but I won't block this PR on this.